### PR TITLE
Don't write output if nothing goes wrong

### LINKED ISSUE #16: K8up compatibility
## Context
To be used as part of K8up based cluster backups, some changes must be made:

- No log messages may be written to STDOUT entirely
- Log messages to STDERR will cause the K8up backup to fail

Hence the following change is proposed:

- Implement a custom helper function for logging
- In said function, check whether `$DEBUG` is set, and if so, emmit the message to STDOUT
- Always use said function to emmit log output
- Exceptions: Actual errors, they may just go to STDERR

## Alternatives
- Instead of setting `$DEBUG`, use `$LOG_DEST` which defaults to `/dev/null` which can be used to set the logging target (File, `/dev/stderr`, ...)

### PR DESCRIPTION
**Pull request by mhutter**
Writing to stderr will cause the K8up backup to fail.

Fixes #16

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Link this PR to related issues.

### CODE REVIEW

**Review by TheBigLee — COMMENTED** [review]
Since we are already touching this. Probably best to run this through shellcheck.com and fix the issues that pop up there.

**Comment by mhutter** [comment]
> Probably best to run this through shellcheck.com and fix the issues that pop up there

I considered this but the list is pretty long so I didn't do it :P

**Review by TheBigLee — APPROVED** [review]
Ok. Looks good to me

File: dump-objects [hunk 1]
```diff
@@ -240,7 +244,7 @@ fetch_objects() {
     local errprefix=
 
     if verbose; then
```
Comment (Kidswiss) [inline dump-objects:246]: To get this output I'd have to set verbose and debug now? Wouldn't it make sense to deprecate the verbose flag?

File: dump-objects [hunk 2]
```diff
@@ -376,12 +380,12 @@ for i in "${kinds[@]}"; do
 
   if [[ "$i" == */* ]]; then
     if verbose; then
```
Comment (Kidswiss) [inline dump-objects:382]: Same here.

**Comment by Kidswiss** [comment]
Had two smaller remarks, the rest LGTM

**Comment by mhutter** [comment]
@Kidswiss Yes I considered this as well, but since the `verbose` flag would produce some ADDITIONAL output I left it as is. I'm fine with both.

**Comment by Kidswiss** [comment]
No hard feelings, both is fine.

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**